### PR TITLE
Add more functional tests

### PR DESCRIPTION
**Pull request by dougbu**
- #180 part 1 of n
- extend first (Dropbox) functional tests
  - rename test class to match sample project; now `DropboxCoreReceiverTest`
  - add `DropboxCoreReceiverTest.WebHookAction_NonGetOrPost_IsNotAllowed(...)`
- prepare other (non-Dropbox) samples for functional testing
  - target frameworks specified in `$(StandardTestWebsiteTfms)` property
  - refactor `Program`s to meet Mvc.Testing expectations
  - add default secret key to appsettings.json files; add this file if necessary
  - fix BitbucketStronglyTypedCoreReceiver to compile with .NET 4.6.1 and to run validations
- add functional tests of remaining (non-Dropbox) samples
  - does not yet include tests requiring a non-empty request body

nits: VS removed some BOMs

### CODE REVIEW

**Comment by dougbu** [comment]
@mkArtakMSFT since this PR changes only samples and tests, does it need a `SHP: Approval pending` label?

File: samples/BitbucketStronglyTypedCoreReceiver/BitbucketStronglyTypedCoreReceiver.csproj [hunk 1]
```diff
@@ -1,10 +1,11 @@
 <Project Sdk="Microsoft.NET.Sdk.Web">
   <PropertyGroup>
-    <TargetFramework>netcoreapp2.0</TargetFramework>
+    <TargetFrameworks>$(StandardTestWebsiteTfms)</TargetFrameworks>
   </PropertyGroup>
 
   <ItemGroup>
     <PackageReference Include="Microsoft.AspNetCore" Version="$(MicrosoftAspNetCorePackageVersion)" />
+    <PackageReference Include="Microsoft.AspNetCore.Mvc.DataAnnotations" Version="$(MicrosoftAspNetCoreMvcDataAnnotationsPackageVersion)" />
```
Comment (pranavkm) [inline samples/BitbucketStronglyTypedCoreReceiver/BitbucketStronglyTypedCoreReceiver.csproj]: Why is this required - desktop support?

Comment (dougbu) [reply samples/BitbucketStronglyTypedCoreReceiver/BitbucketStronglyTypedCoreReceiver.csproj]: The main WebHooks projects only depend on Microsoft.AspNetCore.Mvc. So, yes, need more when running on full framework.

Comment (dougbu) [reply samples/BitbucketStronglyTypedCoreReceiver/BitbucketStronglyTypedCoreReceiver.csproj]: Oops, Mvc**.Core**

**Comment by dougbu** [comment]
0eaf1540b8
